packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -115,21 +188,40 @@ - (instancetype)initWithURL:(NSURL*)url frameUpdater:(FLTFrameUpdater*)frameUpda
           if (_disposed) return;
           if ([videoTrack statusOfValueForKey:@"preferredTransform" error:nil] ==
               AVKeyValueStatusLoaded) {
-            dispatch_async(dispatch_get_main_queue(), ^{
-              [_player replaceCurrentItemWithPlayerItem:item];
-            });
+            CGSize size = videoTrack.naturalSize;
+            NSLog(@"preferredTransform width, height: %f, %f", size.width, size.height);
+
+            // Rotate the video by using a videoComposition and the preferredTransform
+            _preferredTransform = [self fixTransform:videoTrack];
+            // Note:
+            // https://developer.apple.com/documentation/avfoundation/avplayeritem/1388818-videocomposition
+            // Video composition can only be used with file-based media and is not supported for
+            // use with media served using HTTP Live Streaming.
+            AVMutableVideoComposition* videoComposition =
+                [self getVideoCompositionWithTransform:_preferredTransform
```

Comment:
It is hard for me to believe that we need make a videoComposition to transform the video. That would mean everyone using AVPlayer for playing general videos have to do this. There should be a simpler setting somewhere.
Does `setAffineTransform:` not help like in https://stackoverflow.com/a/42302144 ?

Reply:
I believe, the React plugin doesn't have the same rotation issue as they are using an `AVPlayerLayer` (https://github.com/react-native-community/react-native-video/blob/master/ios/Video/RCTVideo.m#L1187) which handles the necessary rotations behind the scenes. Generally, using an `UIView` to display videos handles the transforms itself. 

We are however, using the output of an`AVPlayerItem` directly and forwarding it to a GL texture. Doing this does not apply any transforms whatsoever: https://stackoverflow.com/questions/12500408/can-i-use-avfoundation-to-stream-downloaded-video-frames-into-an-opengl-es-textu and https://github.com/flutter/plugins/blob/3cba0f6cf02d79a4fc67330691228c62d487b14e/packages/video_player/ios/Classes/VideoPlayerPlugin.m#L258

Please take this with a grain of salt as this is something I researched a few months ago and I no longer remember all details.

Another option would be to do the transform with flutter itself, but I'm not sure that would be any better that using a video composition. These are the only 2 options I could think of, but happy to be corrected if there is better way of doing this.